packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -135,6 +133,25 @@ interface ISerializerEvent extends IEvent {
 	(event: "saved", listener: (dirty: boolean) => void): void;
 }
 
+class RefreshPromiseTracker {
+	public get hasPromise(): boolean {
+		return this.#promise !== undefined;
+	}
+	public get Promise(): Promise<number> | undefined {
+		return this.#promise;
+	}
+	constructor(private readonly catchHandler: (error: Error) => void) {}
+
+	#promise: Promise<number> | undefined;
+	setPromise(p: Promise<number>): void {
+		if (this.hasPromise) {
+			throw new Error("Cannot set promise while promise exists");
+		}
+		this.#promise = p;
+		p.catch(this.catchHandler).finally(() => { this.#promise = undefined; });
```

Comment:
Should this unset the `#promise` in the case of error?  Vs. should the error "jam" the refresher?

Either way probably add a comment about the intended behavior in the case of error.

Reply:
i don't really understand this comment. `finally` will clear the promise regardless of error or success